### PR TITLE
enhancement/9 add optimization, add edit user, add delete user

### PR DESCRIPTION
**Pull request by nikita-parkhomenko**
- Add functionality to action button
- Enhancement header styles
- Add memoization for all methods
- Add edit-user page with dynamic id params
- Add errors handling
- Implement initialValues for edit-user fields
- Fetch user for edit from api
- Enhancement initialize saga with conditional api call
- Save edit-user to api calling editUser method
- Add deleteItem saga to delete user from api by id
- Add api call methods to api-service class
- Add dynamic routes to routes file

### CODE REVIEW

File: src/components/action-button.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
 
 // outsource dependencies
-import React from 'react';
+import React, { useCallback } from 'react';
 import { Button } from 'reactstrap';
```
Comment (aaleksandrii) [inline src/components/action-button.js:4]: Line it up

File: src/pages/private/users/saga.js [hunk 2]
```diff
@@ -47,7 +47,20 @@ function * filterItemsSaga({ payload }) {
     yield put({ type: TYPE.META, payload: { disabled: false } });
 }
 
+function * deleteItemSaga(action) {
```
Comment (aaleksandrii) [inline src/pages/private/users/saga.js:50]:
```suggestion
function * deleteItemSaga({ type, payload }) {
```

File: src/services/api-service.js [hunk 3]
```diff
@@ -61,7 +61,9 @@ class ApiService {
             }
         })
             .then(resp => resp.data.content)
-            .then(roles => roles.map(role => ({ label: role.name, value: role.name, id: role.id }) ))
+            .then(roles => roles.map(
+                role => ({ label: role.name, value: role.name, id: role.id, name: role.name })
```
Comment (aaleksandrii) [inline src/services/api-service.js:65]:
```suggestion
.then(roles => roles.map( ({ id, name }) => ({ label: name, value: name, id, name })
```

File: src/components/redux-form-async-select.js [hunk 4]
```diff
@@ -15,11 +15,12 @@ const ReactReduxAsyncSelect = ({ input, label, meta: { error, touched } }) => (
             defaultOptions
             value={input.value}
             onChange={input.onChange}
-            onBlur={() => input.onBlur(input.value)}
-            loadOptions={name => ApiService.loadRoles(name)}
+            onBlur={useCallback(() => input.onBlur(input.value), [input])}
+            loadOptions={useCallback(name => ApiService.loadRoles(name), [])}
```
Comment (aaleksandrii) [inline src/components/redux-form-async-select.js:19]: use getOptionValue, getOptionLabel

File: src/components/redux-form-async-select.js [hunk 5]
```diff
@@ -15,11 +15,12 @@ const ReactReduxAsyncSelect = ({ input, label, meta: { error, touched } }) => (
             defaultOptions
             value={input.value}
             onChange={input.onChange}
-            onBlur={() => input.onBlur(input.value)}
-            loadOptions={name => ApiService.loadRoles(name)}
+            onBlur={useCallback(() => input.onBlur(input.value), [input])}
```
Comment (aaleksandrii) [inline src/components/redux-form-async-select.js:18]: don't use hooks in JSX

File: src/services/api-service.js [hunk 6]
```diff
@@ -78,9 +80,40 @@ class ApiService {
...
+                id,
+            }]
+        })
+    }
+
+    static getUserById(id) {
+        return instance(`admin-service/users/${id}`, {
+            method: "GET",
+        })
+            .then(resp => {
```
Comment (aaleksandrii) [inline src/services/api-service.js:104]: why?

**Review by aaleksandrii — CHANGES_REQUESTED** [review]

File: src/services/api-service.js [hunk 7]
```diff
@@ -78,9 +80,40 @@ class ApiService {
...
+                const { roles, ...otherProps } = resp.data;
+                return {
+                    ...otherProps,
+                    roles: roles.map(role => ({ value: role.name, label: role.name, ...role }))
```
Comment (ichabanivan) [inline src/services/api-service.js:108]: it should be in the select

File: src/services/api-service.js (hunk 3 above)
Comment (ichabanivan) [reply src/services/api-service.js:65]: it should be in the select, and why 4 property?

**Pull request merged**